src/blockdata/transaction.rs

```diff
@@ -888,15 +890,21 @@ impl crate::consensus::encode::Encodable for SubField {
                 Ok(len + nonce.consensus_encode(w)?)
             }
             SubField::MergeMining(ref depth, ref merkle_root) => {
-                len += 0x3u8.consensus_encode(w)?;
-                match depth {
-                    Some(dep) => {
-                        len += VarInt(33).consensus_encode(w)?;
-                        len += dep.consensus_encode(w)?;
-                    }
-                    None => len += VarInt(32).consensus_encode(w)?,
+                // As per monero code after `release-v0.18`, `bool add_mm_merkle_root_to_tx_extra(...)`
+                {
+                    // Tag (u8)
+                    len += 0x3u8.consensus_encode(w)?;
+                    // Data size (u8)
+                    let mut writer_temp = vec![];
+                    let depth_var_int_size = depth.consensus_encode(&mut writer_temp)?;
+                    let data_length = 32 /*hash*/ + depth_var_int_size.to_le_bytes()[0];
```

Comment:
could we just use as u8 here?

```suggestion
                    let data_length = 32 /*hash*/ + depth_var_int_size as u8;
 ```